### PR TITLE
Moving code from Flame

### PR DESCRIPTION
**Pull request by erickzanardo**
Moving the code of the tiled support from the Flame repository.

Note, I used this movement, and added a plain, unaware of the component system class for our tiled support, and now the TiledComponent uses this class, so now people that want to use tiled, but don't want to use the component system has an easy alternative.

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Lgtm, just add a more meaningful changelog message on the CHANGELOG file ;)

File: example/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Flame Tiled example
+
+description: Simple project to showcase the usage of flame_tiled
```
Comment (luanpotter) [inline example/README.md]: I don't think we need `description:` on a markdown right?

File: test/tiled_test.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flame/flame.dart';
+import 'package:flutter/services.dart' show CachingAssetBundle;
+import 'package:test/test.dart';
+import '../lib/tiled_component.dart';
+
+void main() {
+  test('my first widget test', () async {
```
Comment (luanpotter) [inline test/tiled_test.dart]: `my first widget test`?

File: test/tiled_test.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    await Flame.init(bundle: TestAssetBundle());
+    final tiled = TiledComponent('x', 16);
+    await tiled.future;
+    expect(1, equals(1));
```
Comment (luanpotter) [inline test/tiled_test.dart:15]: why not assert a random property inside `tiled` to make sure the `future` was loaded? (eg what does it load, check tiled.prop is not null)

**Review by luanpotter — APPROVED** [review]
Just a few nits, but otherwise LGTM

**Review by renancaraujo — APPROVED** [review]

File: example/README.md (hunk 1 above)
Comment (erickzanardo) [reply example/README.md]: Nope, gonna remove it

File: test/tiled_test.dart (hunk 2 above)
Comment (erickzanardo) [reply test/tiled_test.dart]: O copied this from the original repository :P

But I am gonna change this

File: test/tiled_test.dart (hunk 3 above)
Comment (erickzanardo) [reply test/tiled_test.dart:15]: Again, I copied this from the original repository, I just made it worked in the new structure, I bet the old reasinigng was that no errors were throwed.

**Pull request merged**